resources/migrations/000_migrations.yaml

```diff
@@ -12286,6 +12286,113 @@ databaseChangeLog:
+                  name: dashboard_id
+                  type: int
+                  remarks: The related dashboard
+            tableName: emitter
+
+  - changeSet:
+      id: v45.00-001
+      author: snoe
+      comment: Added 0.45.0 - writeback add action_id to emitter
+      changes:
+        - addColumn:
+            columns:
+              - column:
+                  name: action_id
+                  type: int
+                  remarks: The related action
+            tableName: emitter
+
+  - changeSet:
+      id: v45.00-002
+      author: snoe
+      comment: Added 0.45.0 - writeback add fk for dashboard_id to emitter
+      changes:
+        - addForeignKeyConstraint:
+            baseTableName: emitter
+            baseColumnNames: dashboard_id
+            referencedTableName: report_dashboard
+            referencedColumnNames: id
+            constraintName: fk_emitter_ref_dashboard_id
+            onDelete: CASCADE
+
+  - changeSet:
+      id: v45.00-003
+      author: snoe
+      comment: Added 0.45.0 - writeback add fk for action_id to emitter
+      changes:
+        - addForeignKeyConstraint:
+            baseTableName: emitter
+            baseColumnNames: action_id
+            referencedTableName: action
+            referencedColumnNames: id
+            constraintName: fk_emitter_ref_action_id
+            onDelete: CASCADE
+
+  - changeSet:
+      id: v45.00-004
+      author: snoe
+      comment: Added 0.45.0 - writeback Move foreign keys from emitter_action and dashboard_emitter
+      changes:
+        - sql:
+              sql: >-
+                update emitter set
+                action_id = (select action_id from emitter_action where emitter_id = id),
+                dashboard_id = (select dashboard_id from dashboard_emitter where emitter_id = id);
+
+  - changeSet:
+      id: v45.00-005
+      author: snoe
+      comment: Added 0.45.0 - writeback Make emitter action_id not nullable
+      changes:
+        - addNotNullConstraint:
+            columnDataType: int
+            tableName: emitter
+            columnName: action_id
+
+  - changeSet:
+      id: v45.00-006
+      author: snoe
+      comment: Added 0.45.0 - writeback Make emitter dashboard_id not nullable
+      changes:
+        - addNotNullConstraint:
+            columnDataType: int
+            tableName: emitter
+            columnName: dashboard_id
+
+  - changeSet:
+      id: v45.00-007
+      author: snoe
+      comment: Added 0.45.0 - writeback drop card_emitter
+      changes:
+        - dropTable:
+            tableName: card_emitter
+
+  - changeSet:
+      id: v45.00-008
+      author: snoe
+      comment: Added 0.45.0 - writeback drop dashboard_emitter
+      changes:
+        - dropTable:
+            tableName: dashboard_emitter
+
+  - changeSet:
+      id: v45.00-009
+      author: snoe
+      comment: Added 0.45.0 - writeback drop emitter_action
+      changes:
+        - dropTable:
+            tableName: emitter_action
```

Comment:
Question: These drop tables can't rollback manually, is that a concern?

Reply:
Question: should the v44 writeback migrations move to v45? They were not released as part of 44 but the writeback demo site did run them.

Reply:
Yeah, we should move them all to 45. You can add conditions to skip them if their old numbers have already been ran. That can be done as a separate PR tho. It might actually make sense to tackle that first and then rebase this PR off of that one